File: .agy/skills/jbr-board-scraper/scripts/scrape_news.py
import asyncio
import json
from crawl4ai import AsyncWebCrawler

# Target URLs for staffing news
TARGET_URLS = [
    "https://www.staffingindustry.com/",
    "https://recruitingdaily.com/"
]

async def scrape_news():
    results = []
    print("Initializing Crawl4AI...")
    async with AsyncWebCrawler(verbose=True) as crawler:
        for url in TARGET_URLS:
            print(f"Crawling {url}...")
            try:
                result = await crawler.arun(url=url)
                # Extract markdown content and basic metadata
                results.append({
                    "url": url,
                    "content": result.markdown[:2000] if result.markdown else "No content extracted",
                    "success": result.success
                })
            except Exception as e:
                print(f"Error crawling {url}: {e}")
    
    # Output as JSON for the agent to parse
    with open("scraper_output.json", "w", encoding="utf-8") as f:
        json.dump(results, f, indent=2)
    print("Scraping complete. Results saved to scraper_output.json")

if __name__ == "__main__":
    asyncio.run(scrape_news())

File: .agy/skills/jbr-board-scraper/scripts/scrape_news.js
const puppeteer = require('puppeteer');
const fs = require('fs');

const TARGET_URLS = [
    "https://www.staffingindustry.com/",
    "https://recruitingdaily.com/"
];

async function scrapeNews() {
    console.log("Initializing Puppeteer...");
    const browser = await puppeteer.launch({ headless: 'new' });
    const page = await browser.newPage();
    const results = [];

    for (const url of TARGET_URLS) {
        console.log(`Crawling ${url}...`);
        try {
            await page.goto(url, { waitUntil: 'networkidle2', timeout: 30000 });
            // Extract the main body text to simulate raw extraction
            const textContent = await page.evaluate(() => document.body.innerText.substring(0, 3000));
            results.push({
                url: url,
                content: textContent,
                success: true
            });
        } catch (e) {
            console.error(`Error crawling ${url}: ${e.message}`);
            results.push({ url: url, success: false, error: e.message });
        }
    }
    
    await browser.close();
    fs.writeFileSync("scraper_output.json", JSON.stringify(results, null, 2));
    console.log("Scraping complete. Results saved to scraper_output.json");
}

scrapeNews();
